websites/K/KOLEO/presence.ts

```diff
@@ -0,0 +1,1003 @@
+				const tempTypeOfTicket = document.querySelector(
+					".order-summary-info__content"
+				);
+				if (tempTypeOfTicket) typeOfTicket = tempTypeOfTicket.textContent;
+				const tempCostOfOrder = document.querySelector(
+					".order-summary-total__price"
+				);
+				if (tempCostOfOrder) costOfOrder = tempCostOfOrder.textContent;
+				paymentMethod = document.querySelector(
+					".tile-radio--is-checked .payment-method-radio__control span"
+				);
+			} else if (pathname.startsWith("/confirm")) {
+				const startStation = document.querySelector("li.koleoicon-arrow_right"),
+					endStation = document.querySelector("li.koleoicon-arrow_left");
+				if (startStation && endStation)
+					titleOfOrder = `${startStation.textContent} - ${endStation.textContent}`;
+				const tempDateOfTransport =
+					document.querySelector("li.koleoicon-clock");
+				if (tempDateOfTransport)
+					dateOfTransport = tempDateOfTransport.textContent;
+				const tempTypeOfTicket = document.querySelector(
+					".koleoicon-offer.ticket-type-info"
+				);
+				if (tempTypeOfTicket)
+					typeOfTicket = tempTypeOfTicket.textContent.toLowerCase();
+				const tempCostOfOrder =
+					document.querySelector(".payment-sum") ||
+					document.querySelector(".sum-to-pay");
+				if (tempCostOfOrder) costOfOrder = tempCostOfOrder.textContent;
+				paymentMethod = document.querySelector("li.active");
+			}
+			if (!titleOfOrder || !dateOfTransport || !typeOfTicket || !costOfOrder) {
+				presenceData.details = "Ładuje zamówienie...";
+				presenceData.smallImageText = "Ładuje zamówienie...";
+				presenceData.smallImageKey = Assets.Buy;
+				return presence.setActivity(presenceData);
+			}
+
+			if (paymentMethod) {
+				if (
+					paymentMethod.classList.contains("payment-method-wallet__name") ||
+					paymentMethod.classList.contains("payment-koleo-account")
+				)
+					title = "środków na koncie KOLEO";
+				else if (
+					paymentMethod.classList.contains("payment-method-blik__name") ||
+					paymentMethod.classList.contains("payment-blik")
+				)
+					title = "BLIKa";
+				else if (
+					paymentMethod.classList.contains("payment-method-card__name") ||
+					paymentMethod.classList.contains("payment-card")
+				)
+					title = "karty płatniczej";
+				else if (paymentMethod.classList.contains("payment-trans"))
+					title = "szybkiego przelewu";
+			}
+
+			if (!privacySetting) {
+				presenceData.details = `Kupuje bilet ${typeOfTicket}.`;
+				if (title) presenceData.state = `Za pomocą ${title}.`;
+				else {
+					presenceData.state = `Z ${titleOfOrder.split(" - ")[0]} - Do ${
+						titleOfOrder.split(" - ")[1]
+					} - ${dateOfTransport} - Za cenę ${costOfOrder}.`;
+				}
+			} else presenceData.details = "Kupuje bilet.";
+			presenceData.smallImageText = "Kupuje bilet...";
+			presenceData.smallImageKey = Assets.Buy;
+		} else if (pathname.startsWith("/ticket/")) {
+			const tickets = document.querySelectorAll(".ticket");
+			if (!tickets) {
```

Comment:
This condition is always `false`, since tickets is a `NodeList` object (a truthy value)

Reply:
This is not resolved yet.
```suggestion
			if (tickets.length === 0) {
```